src/ProductConstructionService/ProductConstructionService.FeedCleaner/FeedCleanerJob.cs

```diff
@@ -0,0 +1,161 @@
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+using PackagesInReleaseFeeds = System.Collections.Generic.Dictionary<string, System.Collections.Generic.Dictionary<string, System.Collections.Generic.HashSet<string>>>;
+
+namespace ProductConstructionService.FeedCleaner;
+
+public class FeedCleanerJob
+{
+    private readonly IServiceProvider _serviceProvider;
+    private readonly IAzureDevOpsClient _azureDevOpsClient;
+    private readonly IOptions<FeedCleanerOptions> _options;
+    private readonly Microsoft.Extensions.Logging.ILogger<FeedCleanerJob> _logger;
+
+    public FeedCleanerJob(
+        IServiceProvider serviceProvider,
+        IAzureDevOpsClient azureDevOpsClient,
+        IOptions<FeedCleanerOptions> options,
+        Microsoft.Extensions.Logging.ILogger<FeedCleanerJob> logger)
+    {
+        _serviceProvider = serviceProvider;
+        _azureDevOpsClient = azureDevOpsClient;
+        _options = options;
+        _logger = logger;
+    }
+
+    private FeedCleanerOptions Options => _options.Value;
+
+    public async Task CleanManagedFeedsAsync()
+    {
+        if (!Options.Enabled)
+        {
+            _logger.LogInformation("Feed cleaner service is disabled in this environment");
+            return;
+        }
+
+        _logger.LogInformation("Loading packages in release feeds...");
+
+        PackagesInReleaseFeeds packagesInReleaseFeeds = await GetPackagesForReleaseFeedsAsync();
+
+        _logger.LogInformation("Loaded {versionCount} versions of {packageCount} packages from {feedCount} feeds",
+            packagesInReleaseFeeds.Sum(feed => feed.Value.Sum(package => package.Value.Count)),
+            packagesInReleaseFeeds.Sum(feed => feed.Value.Keys.Count),
+            packagesInReleaseFeeds.Keys.Count);
+
+        foreach (var azdoAccount in Options.AzdoAccounts)
+        {
+            _logger.LogInformation("Processing feeds for {account}...", azdoAccount);
+
+            List<AzureDevOpsFeed> allFeeds;
+            try
+            {
+                allFeeds = await _azureDevOpsClient.GetFeedsAsync(azdoAccount);
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, "Failed to get feeds for account {azdoAccount}", azdoAccount);
+                continue;
+            }
+
+            List<AzureDevOpsFeed> managedFeeds = allFeeds
+                .Where(f => Regex.IsMatch(f.Name, FeedConstants.MaestroManagedFeedNamePattern))
+                .Shuffle()
+                .ToList();
+
+            _logger.LogInformation("Found {totalCount} feeds for {account}. Will process {count} matching feeds",
+                allFeeds.Count,
+                azdoAccount,
+                managedFeeds.Count);
+
+            int feedsCleaned = 0;
+
+            Parallel.ForEach(
+                managedFeeds,
+                new ParallelOptions
+                {
+                    MaxDegreeOfParallelism = 5
```

Comment:
Don't we want this to be configurable instead of hardcoded?

Reply:
I think we don't really care